plugins/targetlocker/inmemory/inmemory.go

```diff
@@ -95,30 +111,48 @@ func broker(lockRequests, unlockRequests, checkLocksRequests <-chan request, don
 			}
 			req.err <- lockErr
 		case req := <-unlockRequests:
+			if err := validateRequest(req); err != nil {
+				req.err <- fmt.Errorf("unlock request: %w", err)
+			}
 			log.Debugf("Requested to transactionally unlock %d targets: %v", len(req.targets), req.targets)
+			var unlockErr error
 			for _, t := range req.targets {
-				if _, ok := locks[*t]; ok {
-					delete(locks, *t)
+				if l, ok := locks[*t]; ok {
```

Comment:
previously, in unlock requests, I was just deleting the lock, if found. This is wrong when the lock is not owned by the same job ID, so now job IDs are checked